### PR TITLE
WIP: Create outline option

### PR DESCRIPTION
**Pull request by aeslaughter**
@crswong888 This is the start of the outline command. My explanation wasn't all that great and I also gave you a bit of misinformation, sorry for that trouble.
(refs #15232)

### CODE REVIEW

**Comment by crswong888** [comment]
@aeslaughter No worries! It seems that we faced a novel type problem with MooseDocs that didn't have an obvious solution.

Also, no trouble all. This was an opportunity for me to start learning this code, and only now I can understand exactly what you've done here. Plus, it seems like you inherited a bit from my noob version.

File: python/MooseDocs/extensions/content.py [hunk 1]
```diff
@@ -162,6 +163,25 @@ def createToken(self, parent, info, page):
...
+class ContentOutlineCommand(command.CommandComponent):
+    COMMAND = 'content'
+    SUBCOMMAND = 'outline'
+
+    @staticmethod
+    def defaultSettings():
+        settings = command.CommandComponent.defaultSettings()
+        settings['location'] = (None, "The markdown content directory to build outline.")
+        settings['level'] = (2, 'Heading level(s) to display.')
+        return settings
```
Comment (crswong888) [inline python/MooseDocs/extensions/content.py:175]: I think it would be nice to have the `'hide'` setting here, no?

Comment (aeslaughter) [reply python/MooseDocs/extensions/content.py:175]: It wouldn't hurt.

**Comment by aeslaughter** [comment]
@crswong888 Do you want to finish this implementation or is it something you want me to do?

**Comment by crswong888** [comment]
@aeslaughter I want to give it a try, because I want to learn the rendering part.

**Pull request closed without merging**